src/heretic/g_game.c

```diff
@@ -307,9 +315,12 @@ void G_BuildTiccmd(ticcmd_t *cmd, int maketic)
 
     strafe = gamekeydown[key_strafe] || mousebuttons[mousebstrafe]
         || joybuttons[joybstrafe];
-    speed = joybspeed >= MAX_JOY_BUTTONS
-         || gamekeydown[key_speed]
-         || joybuttons[joybspeed];
+
+    // [crispy] when "always run" is active,
+    // pressing the "run" key will result in walking
+    speed = key_speed >= NUMKEYS
```

Comment:
I copied this over from Doom since I'd like to have the same behaviour across the two games for these kinds of features. Should it be removed from the Doom side as well?

Reply:
No, because in Doom there is already the `NUMKEYS` condition in this part of the code. We are not here to introduce new conditions, just to modify the existing checks.